packages/lib/src/artifacts/ZosContract.ts

```diff
@@ -0,0 +1,49 @@
+import BN from 'bignumber.js';
+import ZWeb3 from './ZWeb3';
+import { getSolidityLibNames, hasUnlinkedVariables } from '../utils/Bytecode';
+import { Contract, TransactionObject } from 'web3-eth-contract';
+import { TransactionReceipt } from 'web3/types';
+import Contracts, { ZosContractSchema } from './Contracts';
+import _ from 'lodash';
+
+export default class ZosContract {
+  public schema: ZosContractSchema;
+
+  constructor(schema: any) {
+    this.schema = schema;
+  }
+
+  public async deploy(args: any[] = [], options: any = {}): Promise<Contract> {
+    if(!this.schema.linkedBytecode) throw new Error(`${this.schema.contractName} bytecode contains unlinked libraries.`);
+    const contract = ZWeb3.contract(this.schema.abi, null, await Contracts.getDefaultTxParams());
+    const self = this;
+    return new Promise(function(resolve, reject) {
+      const tx = contract.deploy({data: self.schema.linkedBytecode, arguments: args});
+      const zosData: any = {};
+      tx.send({ ...options })
+        .on('error', (error) => reject(error))
+        .on('receipt', (receipt) => zosData.deploymentTransactionReceipt = receipt)
+        .on('transactionHash', (hash) => zosData.deploymentTransactionHash = hash)
+        .then((instance) => {
+          instance.zosData = zosData;
```

Comment:
If we are just storing txReceipt and txHash, how about storing it in a "deployment" node, instead of naming it zosData?

Reply:
Yeah, that'd make sense. However "zosData" is more explicit in the sense that the mere name suggests that this was something injected by zOS. People could confuse "deployment" as part of Web3 Contract's api. Also, we might want to inject additional data in the future that is not only related to deployment. I suggest:
```
zosData: {
  deployment: {...}
}
```

Reply:
Personally, I'd prefer to see it listed as just `deployment`, without the `zosData` step. I think that the risk of colliding with a future extension to the web3 contract is low.